saleor/graphql/product/bulk_mutations/products.py

```diff
@@ -705,22 +705,54 @@ def perform_mutation(cls, root, info, **data):
             warehouses = cls.get_nodes_or_error(
                 warehouse_ids, "warehouse", only_type=Warehouse
             )
-            cls.update_or_create_variant_stocks(variant, stocks, warehouses)
+            cls.update_or_create_variant_stocks(variant, stocks, warehouses, info)
 
         variant = ChannelContext(node=variant, channel_slug=None)
         return cls(product_variant=variant)
 
     @classmethod
     @traced_atomic_transaction()
-    def update_or_create_variant_stocks(cls, variant, stocks_data, warehouses):
+    def update_or_create_variant_stocks(cls, variant, stocks_data, warehouses, info):
         stocks = []
         for stock_data, warehouse in zip(stocks_data, warehouses):
-            stock, _ = warehouse_models.Stock.objects.get_or_create(
+            stock, created = warehouse_models.Stock.objects.get_or_create(
                 product_variant=variant, warehouse=warehouse
             )
+            stocks.append(
+                {
+                    "stock": stock,
+                    "created": created,
+                    "prev_quantity": stock.quantity,
+                    "current_quantity": stock_data["quantity"],
+                }
+            )
             stock.quantity = stock_data["quantity"]
-            stocks.append(stock)
-        warehouse_models.Stock.objects.bulk_update(stocks, ["quantity"])
+        stocks_to_update = [s["stock"] for s in stocks]
+        warehouse_models.Stock.objects.bulk_update(stocks_to_update, ["quantity"])
+        cls._run_product_variant_stock_exists_webhook(info, stocks)
```

Comment:
We don't need to pass the entire `info` object as we only need the plugins manager. We could pass `info.context.plugins` instead of `info` and in  
`_run_product_variant_stock_exists_webhook`,
`_run_product_variant_stock_changed_webhook` 
receive `(cls, manager, updated_stocks)`